test/Common/TestUtils.cs

```diff
@@ -149,5 +149,14 @@ public static void Retry(Action action, int retryCount = 3, int waitDurationMs =
                 }
             }
         }
+
+        /// <summary>
+        /// Removes spaces, tabs and new lines from the JSON string.
+        /// </summary>
+        /// <param name="jsonStr">The json string for trimming</param>
+        public static string TrimJSONString(string jsonStr)
+        {
+            return jsonStr.Trim().Replace(" ", "").Replace(Environment.NewLine, "");
```

Comment:
I would suggest trying out [Assert.Equivalent](https://github.com/xunit/xunit/issues/1604) (and then parse these into actual JObjects) - you'll need to update to use the 2.4.2-pre.12 version of xunit for that.

Reply:
I tried that but even then it fails since the javascript properties begin with lowercase and c# properties with caps.

Reply:
You can lowercase the JSON string before converting it in to an object

Reply:
if casing is needed either way then can I leave what I've instead of updating it?

Reply:
What do the failures look like? Is it clear where exactly the difference is? The idea behind using the actual objects is it allows for a much more detailed error since it knows exactly what properties didn't match. But if the error message message for strings is clear enough I guess it's fine...

But if you do change the name to `CleanJsonString` - "trimming" is a well-known term that refers specifically to remove values on the ends of a string and so the fact that this is also removing all spaces wouldn't fit the name of the method as it is currently.

Reply:
Assert.Equal() Failure
Expected: [{"ProductID":1,"Name":"test","Cost":100}]
Actual:   [
  {
    "ProductId": 1,
    "Name": "test",
    "Cost": 100
  }
]
So unless I lowercase and check it won't work since ProductID and ProductId are different.